src/coreComponents/constitutive/solid/PorousSolid.hpp

```diff
@@ -285,33 +323,62 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
                                      q,
                                      timeIncrement,
                                      strainIncrement,
-                                     totalStress, // first effective stress increment accumulated
+                                     totalStress, // Rock stress updated with small strain
                                      stiffness );
 
-    // Add the contributions of pressure and temperature to the total stress
-    real64 const biotCoefficient = m_porosityUpdate.getBiotCoefficient( k );
-    real64 const thermalExpansionCoefficient = m_solidUpdate.getThermalExpansionCoefficient( k );
+    // Add the contributions of temperature increment to the rock stress (effective stress).
+    real64 dThermalExpansionCoefficient_dTemperature;
+    real64 thermalExpansionCoefficient;
+
+    if( m_drainedTECTableName == nullptr || m_drainedTECTableName[0] == '\0' )
```

Comment:
what's happening here? This seems hacky and it's in a device function...

Reply:
Yes this is a hack to pass the GPU build after many error trial. It is welcome if you have another solution that is more suitable to you.

Reply:
what is this trying to achieve? Can you present to us at one of the dedicated dev meetings what you are trying to do.